src/caffeine/caffeine.c

```diff
@@ -21,20 +21,28 @@ static gex_Rank_t rank, size;
   const int double_Complex_workaround =4100;
 #endif
 
-void caf_caffeinate(mspace* symmetric_heap, intptr_t* symmetric_heap_start, gex_TM_t* initial_team)
+void caf_caffeinate(mspace* symmetric_heap, intptr_t* symmetric_heap_start, mspace* non_symmetric_heap, gex_TM_t* initial_team)
 {
   GASNET_SAFE(gex_Client_Init(&myclient, &myep, initial_team, "caffeine", NULL, NULL, 0));
 
-  size_t segsz = GASNET_PAGESIZE; // TODO: how big can we make this?
+  size_t segsz = GASNET_PAGESIZE*10; // TODO: how big can we make this?
```

Comment:
The total shared heap size needs to be a user-turnable envvar knob.

To a first approximation I suggest you do something like the following (untested):

```suggestion
  // query largest possible segment GASNet can give us of the same size across all processes:
  size_t max_seg = gasnet_getMaxGlobalSegmentSize(); 
  // impose a reasonable default size
  #ifndef CAF_DEFAULT_SHARED_HEAP_SIZE
  #define CAF_DEFAULT_SHARED_HEAP_SIZE (128*1024*1024) // 128 MiB
  #endif
  size_t default_seg = MIN(max_seg, CAF_DEFAULT_SHARED_HEAP_SIZE);
  // retrieve user preference, defaulting to the above and units of MiB
  size_t segsz = gasnett_getenv_int_withdefault("CAF_SHARED_HEAP_SIZE", 
                                                default_seg, 1024*1024);
  // cap user request to the largest available:     
  // TODO: issue a console warning here instead of silently capping                                           
  segsz = MIN(segsz,max_seg);  
```
The `gasnett_getenv_*` interfaces are [documented here](https://gasnet.lbl.gov/dist-ex/README-tools) and provide the dual benefit of parsing user-provided convenience suffixes (e.g. "1.5 GB") and console reporting their values in GASNET_VERBOSEENV=1 mode.

Reply:
Exactly what I was looking for. Thank you.